### ISSUE TITLE
Common / Useful taskbar settings

### ISSUE THREAD

**Issue by valinet** [valinet/ExplorerPatcher#9]
Here is a list of common registry settings you can make in order to set various options of the taskbar, as you could in Windows 10 from the GUI. Restart Explorer after changing any of them for the effect to take place. To apply, copy the text to a text file, save it as `something.reg` and then open it.

**Disable taskbar buttons grouping**

```
[HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\Explorer]
"NoTaskGrouping"=dword:00000001
```
or
* For the taskbar on the main monitor:
```
[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced]
"TaskbarGlomLevel"=dword:00000002
```
* For the taskbars on other monitors:
```
[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced]
"MMTaskbarGlomLevel"=dword:00000002
```
0 = Always combine, 1 = Combine when taskbar is full, 2 = Never combine

**Enable taskbar on other monitors**

```
[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced]
"MMTaskbarEnabled"=dword:00000001
```

**Behavior of taskbar button on multiple monitors**

```
[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced]
"MMTaskbarMode"=dword:00000002
```
0 = Show taskbar buttons on all taskbars
1 = Show taskbar buttons on main taskbar and taskbar where window is open
2 = Show taskbar buttons on taskbar where window is open

**Small taskbar icons**

```
[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced]
"TaskbarSmallIcons"=dword:00000001
```

**Hide Cortana button**

```
[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced]
"ShowCortanaButton"=dword:00000000
```

**Hide Task View button**

```
[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced]
"ShowTaskViewButton"=dword:00000000
```

**Hide Search box or Search icon**

```
[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Search]
"SearchboxTaskbarMode"=dword:00000000
```
0 = disabled, 1 = show search icon, 2 = show search box

**Change default action of power button in Start (useful in Windows 7-like menus like StartIsBack, OpenShell, also determines the default in the "Shut Down Windows" window (Alt+F4 on the desktop)**

```
[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Search]
"Start_PowerButtonAction"=dword:00000002
```

dword:00000100 = Switch user
dword:00000001 = Sign out
dword:00000010 = Sleep
dword:00000040 = Hibernate
dword:00000002 = Shut down
dword:00000004 = Restart
```

**Comment by Gaurav-Original-ClassicShellTester** [comment]
Could also support switching PowerShell by default to Command Prompt:

Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\Advanced]
"DontUsePowerShellOnWinX"=dword:00000001

Although once PowerShell opens in Windows Terminal, one can easily open Command Prompt in a new tab with 2 more clicks.

**Comment by ErickWei** [comment]
Can we change taskbar buttons width after disable taskbar buttons grouping?
Just same as taskbar buttons grouped?
I can do this with 7++ Taskbar on win10.
it's more efficient

**Comment by valinet** [comment]
Idk, try and see, I don't use 7++ Taskbar. If it does not work, I can't do much about it, the developer should look and try to figure out what changed. I just enable the old taskbar, I don't change about it.

**Comment by Gaurav-Original-ClassicShellTester** [comment]
> Can we change taskbar buttons width after disable taskbar buttons grouping?
> Just same as taskbar buttons grouped?
> I can do this with 7++ Taskbar on win10.
> it's more efficient

7+ Taskbar Tweaker works perfectly once ExplorerPatcher enables the classic Taskbar so everything works as expected. Even StartIsBack++'s Taskbar modifications work (bringing back 32 x 32 sized icons, reducing the margins/padding between taskbar icons and tray icons, the theme engine (ms styles/visual styles) for buttons to get the 3D pressed look. All of 7+ Taskbar Tweaker's advanced options also work.

![image](https://user-images.githubusercontent.com/46422834/131622019-fa5942a7-4ce4-49bc-9c5b-492b744b0234.png)

**Comment by ErickWei** [comment]
> > Can we change taskbar buttons width after disable taskbar buttons grouping?
> > Just same as taskbar buttons grouped?
> > I can do this with 7++ Taskbar on win10.
> > it's more efficient
> 
> 7+ Taskbar Tweaker works perfectly once ExplorerPatcher enables the classic Taskbar so everything works as expected. Even StartIsBack++'s Taskbar modifications work (bringing back 32 x 32 sized icons, reducing the margins/padding between taskbar icons and tray icons, the theme engine (ms styles/visual styles) for buttons to get the 3D pressed look. All of 7+ Taskbar Tweaker's advanced options also work.
> 
> ![image](https://user-images.githubusercontent.com/46422834/131622019-fa5942a7-4ce4-49bc-9c5b-492b744b0234.png)

How to run 7+ Taskbar Tweaker on win11?

**Comment by Gaurav-Original-ClassicShellTester** [comment]
Just get the latest beta version meant for Windows 10: https://rammichael.com/7-taskbar-tweaker The Classic Taskbar on Windows 11 is the same one present in Windows 10 21H1/21H2. Once you enable the classic Taskbar, it should "just work".

If it warns that a particular version isn't supported and asks you for a code to enable experimental support, make a donation to the project and RamMichaeL sends you an experimental support code for beta versions to remove the warning at startup.

**Comment by ErickWei** [comment]
Perfect!
Fisrtly i set NoTaskGrouping=0,and then set **Don't group** on 7++ Taskbar.

See the two Edge buttons....

![TASKBAR1](https://user-images.githubusercontent.com/62826892/131629106-1a8c9ab2-2c0c-4f40-874b-02111e0033b3.png)
![TASKBAR2](https://user-images.githubusercontent.com/62826892/131629119-a7d3d59d-c11e-4272-8942-c2f857afff55.png)

**Comment by valinet** [comment]
@Gaurav-Original-ClassicShellTester It is:
```
[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\StartPage]
"Start_MaximumFrequentApps"=dword:00000006
```

**Comment by BraINstinct0** [comment]
The wiki tab on this github repo could be of some use.
like, 
1. Overview
2. Installation
3. Useful Tweaks
4. Known Bugs

**Comment by valinet** [comment]
Yeah, indeed. If you'd like, you can write and organize some stuff in there, make a PR, I will review it and have it integrated in the upstream.

**Comment by BraINstinct0** [comment]
> Just get the latest beta version meant for Windows 10 from its website. The Classic Taskbar on Windows 11 is the same one present in Windows 10 21H1/21H2. Once you enable the classic Taskbar, it should "just work".
> 
> If it warns that a particular version isn't supported and asks you for a code to enable experimental support, make a donation to the project and RamMichaeL sends you an experimental support code for beta versions to remove the warning at startup.

![image](https://user-images.githubusercontent.com/29201475/132329119-a75e43a9-5733-4d37-b080-3ac74efb4ff2.png)
(note that the language setting is in Korean)
Currently the classic taskbar is enabled as below,
![image](https://user-images.githubusercontent.com/29201475/132329289-7d122234-d492-4eb3-b5c9-ec44a3c14258.png)

 but this doesn't work, saying that library couldn't be loaded. :(

**Comment by polygraphene** [comment]
Can we move taskbar to left side of the screen?
I tried to change StuckRects3 registry key, however this key will be written back to default when restart explorer.

**Comment by Mervinj82** [comment]
Hi is there a way to get the new win11 start icon back after using your DLL? Also is there a way to hide labels in the taskbar (so its just the icons like in wins11) without using 7+ Taskbar Tweaker? Thanks

**Comment by Mervinj82** [comment]
> 
> 
> > Hi is there a way to get the new win11 start icon back after using your DLL? Also is there a way to hide labels in the taskbar (so its just the icons like in wins11) without using 7+ Taskbar Tweaker? Thanks
> 
> Have you checked out the GUI? Literally the second and third option in the "Taskbar" section. As for the Start button image, currently the DLL does not offer any way to change that, maybe there are utilities online that help you do this.

Yes I did try these options out. It un-combines the taskbar but the labels (the names/titles of the apps) are still there [see screenshot below]. I'd like to remove them and just leave the icon. Thanks

![Screenshot](https://user-images.githubusercontent.com/92236600/136680550-305d3e16-1d34-4f86-b9d5-b43335604d2f.png)

**Comment by Mervinj82** [comment]
> hat

That would be great. Id even be ok with having the original Windows 11 look but with just the taskbar items un-grouped if you knew of a way or software to do this. Regarding 7+ Taskbar Tweaker, there is only a beta version and it requires an access code which can only be acquired by donating through PayPal or Bitcoin which I do not currently have.

**Comment by Mervinj82** [comment]
> 
> 
> This program does not currently include the functionality you are asking for. Maybe I will be implement it in the future, but since 7+ Taskbar Tweaker is very robust and fairly modular, why not use that?

That would be great. Id even be ok with having the original Windows 11 look but with just the taskbar items un-grouped if you knew of a way or software to do this. Regarding 7+ Taskbar Tweaker, there is only a beta version and it requires an access code which can only be acquired by donating through PayPal or Bitcoin which I do not currently have.

**Comment by Mervinj82** [comment]
I wouldn't know I haven't tried it. I saw the commentators above getting the results im looking (which is to get rid of the labels in the taskbar) using 7+ TT.

**Comment by valinet** [comment]
I don't understand, the beta version on the site seems to work, and I had to enter no code. And it just does not seem to offert the option you mention. Just... you know... try to install the program? It doesn't bite...

**Comment by Mervinj82** [comment]
> 
> 
> I don't understand, the beta version on the site seems to work, and I had to enter no code. And it just does not seem to offert the option you mention. Just... you know... try to install the program? It doesn't bite...

No need to be patronizing my friend. I think it's pointless to download a software if the author says an access code will be required. Maybe he changed it for the newer release who knows. Anyways if it doesn't offer that option then no problem. I'll wait for an update whether is from you updating the DLL or somethingm from microsoft. Here is the post I read from the author (https://rammichael.com/7-taskbar-tweaker-on-windows-11-with-windows-10s-taskbar)

**Comment by valinet** [comment]
I am not patronizing anyone, I was just saying, it seems more logical to me to try all the avenues you could try. Maybe the blog post is outdated. For me, it didn't ask for any code, it just runs and seems to do its job (I added seconds to the clock, for example). I haven't tested anything. Also, did not see a box to enter any code. No problem... yeah, it is not my problem, you asked about this, and I was looking out, trying to understand exactly what you want. And I have yet to fully grasp it, as neither the blog post, nor you provided any evidence of this supposed feature actually working. I was just curious to see how it works. I don't understand why you need to feel attacked, I think it is plenty logical to try things yourself rather than follow some people in the comments which may even have been talking about something else... even if it had an access code, maybe you could type "0000" and it worked. It's always worth giving a shot rather than nothing... in my opinion, at least.

**Comment by valinet** [comment]
> 
> 
> Thanks for your lengthy response but I'm not really here to argue and I appreciate your work and you taking time to try to help. Like you said the feature I wanted is not available as yet... that's no problem, I'll wait for when it is. 👍

Yeah, i don't want to get in an argument, as well. Sorry if I pissed you off, mainly I was trying to help, and still would like to, hopefully I can find some reference and figure out what this is about. Thanks and hopefully this gets implemented somewhere.

**Comment by Mervinj82** [comment]
Apologies likewise. N no im not pissed off. I guess we just misunderstood each other or something but I appreciate u. Basically what I wanted was to have the labels in the taskbar removed and just the icons would remain. Hope Ive explained better this time. That's why I circled them in my screenshot above in my earlier posts. So if possible it might be a great feature to have in your next update. Keep up the good work.

**Comment by Mervinj82** [comment]
> 
> 
> > Thanks for your lengthy response but I'm not really here to argue and I appreciate your work and you taking time to try to help. Like you said the feature I wanted is not available as yet... that's no problem, I'll wait for when it is. +1
> 
> Yeah, I don't want to get in an argument, as well. Sorry if I pissed you off, mainly I was trying to help, and still would like to, hopefully I can find some reference and figure out what this is about. Thanks and hopefully this gets implemented somewhere.

Apologies likewise. N no im not pissed off. I guess we just misunderstood each other or something but I appreciate u. Basically what I wanted was to have the labels in the taskbar removed and just the icons would remain. Hope Ive explained better this time. That's why I circled them in my screenshot above in my earlier posts. So if possible it might be a great feature to have in your next update. Keep up the good work.

**Comment by valinet** [comment]
Yeah, to be honest, my reply was kind of jerky, I admit. Idk why. Now I feel deeply sorry about it. I had quite some bad time in the last few hours, battled with some issues in the code that kept me wake, been sitting in front of the computer screen for a while. I know, it's no excuse, but sometimes it makes one behave weirdly, the tiredness. Anyway, I will look into it eventually, I was trying to remember, I think I played with something like this years ago but don't really remember where or when or with that tool and it pisses me off even more, the fact that I don't remember. Anyhow, thanks for not getting mad on me, and apologies again.

**Comment by Mervinj82** [comment]
Nah don't worry about I've been there before so I get it. But we good. I'll keep an eye on your updates so hopefully I'll catch it if you happen to add the feature. Have a blessed evening man and get some sleep. lol 👍👍

**Comment by Gaurav-Original-ClassicShellTester** [comment]
> Hi is there a way to get the new win11 start icon back after using your DLL? Also is there a way to hide labels in the taskbar (so its just the icons like in wins11) without using 7+ Taskbar Tweaker? Thanks

As far as I know, hiding labels is a Windows setting since Windows 7 (2009). I believe the setting is:
HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced
TaskbarGlomLevel DWORD 0.

That does not require 7+ Taskbar Tweaker. Unfortunately Windows 7 and all later releases were bad enough to merge hiding/showing labels with button combining into 1 setting. So if you want to hide labels but disable forced combining+grouping based on AppID, first set the Windows setting so labels are hidden, restart Explorer.exe/log off-log on/refresh group policy whatever etc and then use 7+TT to disable combining+grouping.

If you donate to 7+ Taskbar Tweaker, you get your experimental support code in your email which will work for all future early/beta releases. If you don't wish to donate, you can wait just a little bit until the non-beta completely free version is out specifically with support for Windows 11. Although I don't know/haven't tested if the current stable release version of 7+ Taskbar Tweaker works with the classic Taskbar on Windows 11, it should since it is compatible with Windows 10 21H1/21H2 already. I generally prefer the latest beta and I have donated to the 7+TT project. Will be donating to ExplorerPatcher too soon.

**Comment by valinet** [comment]
> 
> 
> Nah don't worry about I've been there before so I get it. But we good. I'll keep an eye on your updates so hopefully I'll catch it if you happen to add the feature. Have a blessed evening man and get some sleep. lol 👍👍

Indeed, @Gaurav-Original-ClassicShellTester is right, it works the way he described. Again, the beta version on the web site works just fine for me and does not ask for any code, and I was able to test what he suggested as well. First of all, yeah, set taskbar to "Always combine, hide labels", then it 7TT choose "Don't combine grouped buttons" in "Combining" and even "Don't group" in "Grouping" and then it is basically exactly what you asked for. Yeah, counterintuitive, but it works. Thanks @Gaurav-Original-ClassicShellTester

**Comment by valinet** [comment]
> 
> 
> Hey guys so I just got done trying this and it WORKS. Thanks for taking time out to help. Only thing I'm missing is to change the Start icon cause I rather the Win11 one. I've looked online but haven't come across a way to do this yet. If you know of a way or come across something please put it in this thread here. Thanks again. 👍

I will try to implement this in a future update, but idk how long it will take or when exactly will it arrive. Thanks.

**Comment by BraINstinct0** [comment]
> Hey guys so I just got done trying this and it WORKS. Thanks for taking time out to help. Only thing I'm missing is to change the Start icon cause I rather the Win11 one. I've looked online but haven't come across a way to do this yet. If you know of a way or come across something please put it in this thread here. Thanks again. 👍

IIRC post-Win10 1903, registry hack to change windows start menu icon was blocked, requiring third party tools.(Not sure if it's impossible or just not yet done)
Icon itself seems to be here-`C:\Windows\System32\@WLOGO_48x48.png`.(Win10 logo not 11)

Don't think this would help but leaving this here.

**Comment by BraINstinct0** [comment]
Edit of my previous comment: That was from reference of my (very very dirty) memos on my windows editing; and the problem is, it seems that I mistook resource editing as a registry hack.(I do know the difference now - but back then, well I didn't.)
Quick search done now, what I did(using third party tools) back then was
1) For win7/win8, the [resource edit done on explorer.exe](https://www.gohacking.com/customize-start-menu-windows-7/) does the trick. This 'Resource Hacker' thing is what I used IIRC, which I took as 'registry hack'. *It was not*, my mistake.
2) For Win10, however, doesn't seem to be the case.
